src/org/zaproxy/zap/extension/pscanrulesAlpha/PiiScanner.java

```diff
@@ -0,0 +1,166 @@
+    private static final Pattern Jcb = Pattern.compile("(?:(?:2131|1800|35\\d{3})\\d{11})");
+    private static final Pattern Maestro = Pattern.compile("(?:(?:5[0678]\\d\\d|6304|6390|67\\d\\d)\\d{8,15})");
+    private static final Pattern Mastercard = Pattern.compile("(?:(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12})");
+    private static final Pattern Visa = Pattern.compile("(?:4[0-9]{12})(?:[0-9]{3})?");
+
+    private static Map<String, Pattern> CreditCardPatterns = new HashMap<>();
+
+    static {
+        CreditCardPatterns.put("AmericanExpress", AmericanExpress);
+        CreditCardPatterns.put("DinersClub", DinersClub);
+        CreditCardPatterns.put("Discover", Discover);
+        CreditCardPatterns.put("Jcb", Jcb);
+        CreditCardPatterns.put("Maestro", Maestro);
+        CreditCardPatterns.put("Mastercard", Mastercard);
+        CreditCardPatterns.put("Visa", Visa);
+    }
+
+    public PiiScanner() {
+        //
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        String responseBody = msg.getResponseBody().toString();
+        List<String> candidates = getAllNumberSequences(responseBody);
+        for(String candidate : candidates) {
+            for (String cardType : CreditCardPatterns.keySet()) {
+                Matcher matcher = CreditCardPatterns.get(cardType).matcher(candidate);
+                while (matcher.find()) {
+                    String evidence = matcher.group();
+                    if(validateLuhnChecksum(evidence)) {
+                        raiseAlert(msg, id, evidence, cardType);
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean validateLuhnChecksum(String evidence) {
+        int sum = 0;
+        int parity = evidence.length() % 2;
+        for (int index = 0; index < evidence.length(); index++) {
+            int digit = Integer.parseInt(evidence.substring(index, index + 1));
+            if ((index % 2) == parity) {
+                digit *= 2;
+                if (digit > 9) {
+                    digit -= 9;
+                }
+            }
+            sum += digit;
+        }
+        return (sum % 10) == 0;
+    }
+
+    private void raiseAlert(HttpMessage msg, int id, String evidence, String cardType) {
+        Alert alert = new Alert(getPluginId(), Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH,  getName());
+        alert.setDetail(
+                Constant.messages.getString(MESSAGE_PREFIX + "desc"),
+                msg.getRequestHeader().getURI().toString(),
+                "", //parameter
+                "",                     //attack
+                String.format("Credit Card Type detected: %s", cardType),
```

Comment:
Is this your other/extra info field? If so you could do something like:
In messages.properties:
`pscanalpha.piiscanner.extrainfo = Credit Card Type detected: {0}`
In the scanner:
`Constant.messages.getString(MESSAGE_PREFIX + "extrainfo", cardType);`